cluster/cluster.go

```diff
@@ -140,13 +195,84 @@ func Join(
 		level.Debug(l).Log("msg", "joined cluster", "peers", n)
 	}
 
-	if n > 0 {
-		go p.warnIfAlone(l, 10*time.Second)
-	}
+	p.setInitialFailed(resolvedPeers)
```

Comment:
IMO it would be simpler to initialize `p.failedPeers` with all known peers before calling `ml.Join(...)`.

Reply:
that's WAAAY better :)